osu.Game/Screens/Footer/ScreenFooter.cs

```diff
@@ -32,12 +33,15 @@ public partial class ScreenFooter : OverlayContainer
         private readonly List<OverlayContainer> overlays = new List<OverlayContainer>();
 
         private Box background = null!;
-        private FillFlowContainer<ScreenFooterButton> buttonsFlow = null!;
-        private Container<ScreenFooterButton> removedButtonsContainer = null!;
+        private FillFlowContainer<ScreenFooterButton> leftButtonsFlow = null!;
+        private Container<ShearedButton?> rightButtonContainer = null!;
+        private Container<ScreenFooterButton> removedLeftButtonsContainer = null!;
+        private Container<ShearedButton?> removedRightButtonContainer = null!;
         private LogoTrackingContainer logoTrackingContainer = null!;
 
+        // TODO: this should take the screen's colourProvider instead. hardcode plum for now as daily challenge is the only usage shown to users
         [Cached]
-        private readonly OverlayColourProvider colourProvider = new OverlayColourProvider(OverlayColourScheme.Aquamarine);
+        private readonly OverlayColourProvider colourProvider = new OverlayColourProvider(OverlayColourScheme.Plum);
```

Comment:
This sort of kicking the can down the road does not inspire confidence, what is the plan with this?

Reply:
Using the `ScreenFooter` anywhere other than song select is not something I wanted to happen immediately, I was in the mindset of "we'll cross that bridge when we get to it". But the daily challenge screen came, and I think it's a great idea to try the screen footer there, to resolve the issue of seeing which mods are currently selected, but there are minor concerns like this that I would hope to be left out until follow-up PRs that aim to get the new song select live.

I will try to see if I can address this alongside what I had mentioned in https://github.com/ppy/osu/pull/29165#issuecomment-2257359759 already, though.